toucan_connectors/common.py

```diff
@@ -340,6 +340,45 @@ def convert_to_qmark_paramstyle(query_string: str, params_values: dict) -> tuple
     return re.sub(RE_NAMED_PARAM, "?", query_string), flattened_values
 
 
+def convert_to_numbered_vars_paramstyle(query_string: str, params_values: dict) -> tuple[str, tuple[Any]]:
```

Comment:
It's called the `numeric` param style: https://peps.python.org/pep-0249/#paramstyle

Also, could you please add parametrized unit tests for this function ? If the oracle connectors is discontinued at some point, we won't have any tests left